pkg/reconciler/taskrun/taskrun_test.go

```diff
@@ -1437,6 +1437,305 @@ spec:
+			Limits:   corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("2")},
+		}},
+	}, {
+		name: "both with cpu and memory",
+		task: multiStepsTask,
+		taskRun: parse.MustParseTaskRun(t, `
+metadata:
+  name: foo-taskrun
+  namespace: default
+spec:
+  taskRef:
+    name: foo-task
+  computeResources:
+    requests:
+      cpu: "1"
+      memory: "500Mi"
+    limits:
+      memory: "1Gi"
+`),
+		expectedPodComputeResources: []corev1.ResourceRequirements{{
+			Requests: corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("1"),
+				corev1.ResourceMemory: resource.MustParse("500Mi"),
+			},
+			Limits: corev1.ResourceList{corev1.ResourceMemory: resource.MustParse("1Gi")},
+		}, {
+			Requests: nil,
+			Limits:   corev1.ResourceList{corev1.ResourceMemory: resource.MustParse("1Gi")},
+		}, {
+			Requests: nil,
+			Limits:   corev1.ResourceList{corev1.ResourceMemory: resource.MustParse("1Gi")},
+		}},
+	}, {
+		name: "overwrite StepTemplate resources requirements",
+		task: withStepTemplateTask,
+		taskRun: parse.MustParseTaskRun(t, `
+metadata:
+  name: foo-taskrun
+  namespace: default
+spec:
+  taskRef:
+    name: foo-task
+  computeResources:
+    requests:
+      cpu: "1"
+`),
+		expectedPodComputeResources: []corev1.ResourceRequirements{{
+			Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("1")},
+		}, {
+			Requests: nil,
+		}},
+	}, {
+		name: "overwrite step-level resources requirements",
+		task: stepResourceRequirementsTask,
+		taskRun: parse.MustParseTaskRun(t, `
+metadata:
+  name: foo-taskrun
+  namespace: default
+spec:
+  taskRef:
+    name: foo-task
+  computeResources:
+    requests:
+      cpu: "2"
+`),
+		expectedPodComputeResources: []corev1.ResourceRequirements{{
+			Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("2")},
+		}, {
+			Requests: nil,
+		}},
+	}, {
+		name: "with sidecar resource requirements",
+		task: withSideCarTask,
+		taskRun: parse.MustParseTaskRun(t, `
+metadata:
+  name: foo-taskrun
+  namespace: default
+spec:
+  taskRef:
+    name: foo-task
+  computeResources:
+    requests:
+      cpu: "4"
+`),
+		expectedPodComputeResources: []corev1.ResourceRequirements{{
+			Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("4")},
+		}, {
+			Requests: nil,
+		}, {
+			// the expected sidecar resource requirements
+			Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("750m")},
+			Limits:   corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("1.5")},
+		}},
+	}}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			tc.taskRun.Status.StartTime = &metav1.Time{Time: testClock.Now()}
+
+			d := test.Data{
+				Tasks:    []*v1beta1.Task{tc.task},
+				TaskRuns: []*v1beta1.TaskRun{tc.taskRun},
+			}
+
+			testAssets, cancel := getTaskRunController(t, d)
+			defer cancel()
+			createServiceAccount(t, testAssets, "default", "default")
+
+			// a requeue error is expected by the "running" TaskRun
```

Comment:
this looks like it's testing invalid taskruns-- I don't think that's what we want

Reply:
good catch. let me dig a sec.

Reply:
closed - as rewriting tests under pod.go